modules/core/src/main/scala/lucuma/odb/api/model/targetModel/TargetModel.scala

```diff
@@ -121,29 +131,52 @@ object TargetModel extends TargetModelOptics {
     existence:   Input[Existence]             = Input.ignore,
     name:        Input[NonEmptyString]        = Input.ignore,
     sidereal:    Option[EditSiderealInput]    = None,
-    nonSidereal: Option[EditNonsiderealInput] = None
+    nonSidereal: Option[EditNonsiderealInput] = None,
+    angularSize: Input[AngularSizeInput]      = Input.ignore
   ) {
 
-    def edit(t: TargetModel): ValidatedInput[TargetModel] =
-      editor.map(_.runS(t).value)
-
-    private val editor: ValidatedInput[State[TargetModel, Unit]] =
-      (existence.validateIsNotNull("existence"),
-       name     .validateIsNotNull("name"),
-       sidereal.traverse(_.editor),
-       nonSidereal.traverse(_.editor)
-      ).mapN { (e, n, s, ns) =>
-        for {
-          _ <- TargetModel.existence := e
-          _ <- TargetModel.name      := n
-          _ <- s.fold(State.get[TargetModel].void) { ed =>
-            TargetModel.target.mod_(ed.runS(_).value)
-          }
-          _ <- ns.fold(State.get[TargetModel].void) { ed =>
-            TargetModel.target.mod_(ed.runS(_).value)
-          }
-        } yield ()
-      }
+    val editor: StateT[EitherNec[InputError, *], TargetModel, Unit] = {
+      val validArgs =
+        (
+          existence.validateIsNotNull("existence"),
+          name     .validateIsNotNull("name")
+        ).tupled.toEither
+
+      def editTarget(s: Option[StateT[EitherNec[InputError, *], Target, Unit]]): StateT[EitherNec[InputError, *], TargetModel, Unit] =
+        s.getOrElse(StateT.empty[EitherNec[InputError, *], Target, Unit])
+         .transformS(_.target, (m, t) => TargetModel.target.replace(t)(m))
+
+      val stateAngularSize: StateT[EitherNec[InputError, *], Target, Unit] =
+        angularSize match {
+          // Do nothing
+          case Ignore         =>
+            StateT.empty[EitherNec[InputError, *], Target, Unit]
+
+          // Remove angular size
+          case Unassign       =>
+            (Target.angularSize := Option.empty[AngularSize]).void
+
+          // Create or edit, depending on whether it already exists
+          case Assign(editor) =>
+            StateT.modifyF[EitherNec[InputError, *], Target] { t =>
+              Target
+                .angularSize
+                .modifyF[EitherNec[InputError, *]](
+                  _.fold(editor.create.toEither)(as => editor.edit.runS(as)).map(_.some)
+                )(t)
+            }
+        }
```

Comment:
This pattern of handling `Input` repeats.  I will try to clean it up and abstract it.